packages/TermsAndConditions/Footnote/Footnote.jsx

```diff
@@ -145,46 +148,65 @@ const Footnote = props => {
           !footnoteRef.current.contains(e.target) &&
           e.target.getAttribute('data-tds-id') !== 'footnote-link')
       ) {
-        closeFootnote(e)
+        closeFootnote(e, { returnFocus: false })
       } else if (
         e.type === 'touchstart' &&
         (footnoteRef &&
           e.touches[0].target &&
           !footnoteRef.current.contains(e.touches[0].target) &&
           e.touches[0].target.getAttribute('data-tds-id') !== 'footnote-link')
       ) {
-        closeFootnote(e)
+        closeFootnote(e, { returnFocus: false })
       }
     },
     [closeFootnote]
   )
 
-  const changeHeight = () => {
+  const saveCurrentHeight = () => {
     const oldHeight = listRef.current.offsetHeight
     setBodyHeight(oldHeight)
-    setIsTextVisible(false)
   }
 
-  const handleStyledFootnoteTransitionEnd = async e => {
+  const handleStyledFootnoteTransitionEnd = e => {
     if (e.propertyName === 'transform' && !isOpen) {
       setIsVisible(false)
     }
   }
 
-  const handleTransitionEnd = async e => {
+  const handleTransitionEnd = e => {
     e.persist()
     if (e.propertyName === 'opacity' && !isTextVisible) {
-      await setData({ content, number })
-      const halfPageHeight = window.innerHeight * 0.5 - headerHeight
-      const newHeight = listRef.current.offsetHeight
-      await setBodyHeight(newHeight > halfPageHeight ? halfPageHeight : newHeight)
+      setData({ content, number })
+      if (bodyHeight !== listRef.current.offsetHeight) {
+        // set new height
+        setBodyHeight(listRef.current.offsetHeight)
+      } else {
+        setIsTextVisible(true)
+      }
+    } else {
+      setBodyHeight(listRef.current.offsetHeight)
     }
 
     if (e.propertyName === 'height' && !isTextVisible) {
       setIsTextVisible(true)
     }
   }
 
+  const resetFootnote = () => {
```

Comment:
This is just a question about consistency. Just wondering what was the reason for abstracting `resetFootnote` and `focusHeading` that are used only once in `useEffect` hooks? Some other `useEffect` hooks use functions without abstraction.